src/main.cpp

```diff
@@ -2715,6 +2712,9 @@ bool AcceptBlockHeader(const CBlockHeader& block, CValidationState& state, CBloc
         if (pindexPrev->nStatus & BLOCK_FAILED_MASK)
             return state.DoS(100, error("%s: prev block invalid", __func__), REJECT_INVALID, "bad-prevblk");
     }
+    assert(pindexPrev);
```

Comment:
I'm referring to the fCheckpointsEnabled conditional. Just to the fact that
the checkpoints-checking point in this function moving.